ait/core/server/server.py

```diff
@@ -0,0 +1,255 @@
+import gevent
+import gevent.monkey; gevent.monkey.patch_all()
+from importlib import import_module
+import sys
+
+# import ait
+import ait.core.server
+from stream import PortInputStream, ZMQInputStream, PortOutputStream
+from broker import AITBroker
+from ait.core import log, cfg
+
+
+class AITServer(object):
+    """
+    This server reads and parses config.yaml to create all streams, plugins and handlers
+    specified. It starts all greenlets that run these components and calls on the broker
+    to manage the ZeroMQ connections.
+    """
+    inbound_streams = [ ]
+    outbound_streams = [ ]
+    servers = [ ]
+    plugins = [ ]
+
+    def __init__(self):
+        self.broker = AITBroker()
+
+        self._load_streams()
+        self._load_plugins()
+
+        self.broker.inbound_streams = self.inbound_streams
+        self.broker.outbound_streams = self.outbound_streams
+        self.broker.servers = self.servers
+        self.broker.plugins = self.plugins
+
+        # defining greenlets that need to be joined over
+        self.greenlets = ([self.broker] +
+                           self.broker.plugins +
+                           self.broker.inbound_streams +
+                           self.broker.outbound_streams)
+
+    def wait(self):
+        """
+        Starts all greenlets for concurrent processing.
+        Joins over all greenlets that are not servers.
+        """
+        for greenlet in (self.greenlets + self.servers):
+            log.info("Starting {} greenlet...".format(greenlet))
+            greenlet.start()
+
+        gevent.joinall(self.greenlets)
+
+    def _load_streams(self):
+        """
+        Reads, parses and creates streams specified in config.yaml.
+        """
+        common_err_msg = 'No valid {} telemetry stream configurations found. '
+        specific_err_msg = {'inbound': 'No telemetry will be received (or displayed).',
+                            'outbound': 'No telemetry will be published.'}
```

Comment:
Nit-picky, but we probably want something other than `telemetry` specified here. No `data` or something similar.